src/can2040.c

```diff
@@ -28,11 +29,16 @@
 
 // Helper compiler definitions
 #define barrier() __asm__ __volatile__("": : :"memory")
+#define __DMB() __asm__ __volatile__("dmb 0xF": : :"memory")
+
+// Zephyr already defines these helpers
+// Avoid redefinition warnings
+#ifndef __ZEPHYR__
```

Comment:
I appreciate your platform-agnostic approach, so I can make these macros individually ifndef-guarded to not introduce a Zephyr-specific thing